### ISSUE TITLE
RuntimeError: Job not found on printer

### ISSUE THREAD

**Issue by bararchy** [nehresma/cupsffi#17]
Maybe instead of returning an exception is is better to return this in the status ? 
as in:

```ruby
job.status
=> :job_not_found
```

**Comment by bararchy** [comment]
I see this is also happening when sending remote jobs:
```
[145] pry(main)> job3 = print.print_file("/home/unshadow/Downloads/the-complete-works-of-h-p-lovecraft.pdf")
RuntimeError: Success
```

**Comment by nehresma** [comment]
I agree that this is a better way to handle statuses.  But the API change would break deployments so I haven't made the change.

**Comment by bararchy** [comment]
I see, maybe then add a hash options when sending the print_file to specify what return value I expect ? that would also be cool and can be default for exception which wont heart running deployments

**Comment by nehresma** [comment]
That could work.  Other APIs (particularly I'm thinking of HTTP REST APIs) do something semantically similar where the caller can request the return results in various formats.

This would be a bit different though in that a "format" choice would include a flow control operation -- exception.

I'll do some thinking about this.

**Comment by bararchy** [comment]
Sounds good, thanks for giving this a thought :)
